igvc_navigation/include/igvc_navigation/Graph.h

```diff
@@ -0,0 +1,301 @@
+/**
+The graph contains the nodes and cells that set the foundation of the search
+problem.
+
+In Field D*, Nodes are located on the corners of grid cells (as opposed to the
+center, like in A*, D*, or D*lite).
+
+The Graph object provides an interface for the occupancy grid, as well as nodes.
+To facilitate the search problem, the graph operates under a unit-cell assumption,
+meaning that each cell has side lengths of 1m. The `Resolution` parameter specifies
+the conversion between the unit cell and the actual cell dimensions. i.e. if Resolution
+is 0.2 then each cell actually has side dimensions of 0.2m.
+
+Author: Alejandro Escontrela <aescontrela3@gatech.edu>
+Date Created: December 16, 2018
+*/
+
+#ifndef GRAPH_H
+#define GRAPH_H
+
+#include "Node.h"
+#include "igvc_utils/NodeUtils.hpp"
+
+#include <igvc_msgs/map.h>
+
+#include <cv_bridge/cv_bridge.h>
+#include <cassert>
+#include <cmath>
+#include <limits>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+/**
+A cell represents a grid in the graph. Each cell object contains fields <x,y>
+which denote the cell's location on the graph.
+*/
+struct Cell
+{
+    int x;
+    int y;
+
+    Cell(int x, int y)
+    {
+        this->x = x;
+        this->y = y;
+    }
+
+    Cell(std::tuple<int,int> cell) : Cell(std::get<0>(cell), std::get<1>(cell))
+     {
+     }
+
+     // overloaded assignment operator
+     Cell& operator=(const Cell& other)
+     {
+       this->x = other.x;
+       this->y = other.y;
+
+       return *this;
+     }
+
+     // Cells equal if their corresponding indices are equal
+     bool operator==(const Cell& other) const
+     {
+       return (this->x == other.x) && (this->y == other.y);
+     }
+
+     bool operator!=(const Cell& other) const
+     {
+       return !(*this==other);
+     }
+};
+
+class Graph
+{
+public:
+  /**
+  Prev map is the occupancy grid one timestep ago, it's used to check for changes in the
+  the occupancy grid. Map is the current, most up-to-date occupancy grid.
+  */
+  cv_bridge::CvImagePtr Map;
+
+  Node Start;  // start node in the search problem
+  Node Goal;   // goal node in the search problem
+
+  // Updated cell information is used to update nodes that lie on the 4 corners of
+  // each updated cell. This is reset each time updateGraph is called. Each element
+  // is composed of an <x,y> tuple representing the cell.
```

Comment:
Make up your mind about you're using `/** */` or `//`, and on where you're placing the comment

Reply:
you're right. I'll update